core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -85,6 +90,67 @@
         String getString(String key);
     }
 
+    public static class Access {
+        public final Instant lastAccessTime;
+        public final long accessCount;
+        public final String lastAccessValue;
+        public final Set<StackTraceElement> accessingCode = new HashSet<>();
```

Comment:
There is no logic which would prevent limiting size of this collection.
It will basically end up recording all system properties across the instance, especially once we open up the API to public.
Not a big deal unless users start using this API in their Pipelines and Groovy scripts, then it may lead to infinite explosion due to on-demand class creation in Groovy

Reply:
Actually both this and the concurrency concern don't really really apply since the collection doesn't currently record more than a single entry 🤦